src/ImportPage.js

```diff
@@ -19,27 +22,92 @@ type Props = {|
 |};
 
 export default function ImportPage({onDataImported}: Props) {
+  const processTimeline = useCallback(
+    (events: TimelineEvent[]) => {
+      // Filter null entries and sort by timestamp.
+      // I would not expect to have to do either of this,
+      // but some of the data being passed in requires it.
+      events = events.filter(Boolean).sort((a, b) => (a.ts > b.ts ? 1 : -1));
+
+      if (events.length > 0) {
+        const processedData = preprocessData(events);
+        const processedFlamechart = preprocessFlamechart(events);
+        onDataImported(processedData, processedFlamechart);
+      }
+    },
+    [onDataImported],
+  );
+
+  // ImportPage is show in production build only
   useEffect(() => {
+    if (process.env.NODE_ENV === 'production') return;
     fetch(JSON_PATH)
       .then(res => res.json())
-      .then((events: TimelineEvent[]) => {
-        // Filter null entries and sort by timestamp.
-        // I would not expect to have to do either of this,
-        // but some of the data being passed in requires it.
-        events = events.filter(Boolean).sort((a, b) => (a.ts > b.ts ? 1 : -1));
-
-        if (events.length > 0) {
-          const processedData = preprocessData(events);
-          const processedFlamechart = preprocessFlamechart(events);
-          onDataImported(processedData, processedFlamechart);
-        }
-      });
-  }, []);
+      .then(processTimeline);
+  }, [onDataImported, processTimeline]);
```

Comment:
`onDataImported` isn't used in this hook. We should add an ESLint rule to lint our hook usage instead of manually catching these in PR reviews

```suggestion
  }, [processTimeline]);
```